microsoft/testsuites/vm_extensions/azureperformancediagnostics.py

```diff
@@ -0,0 +1,192 @@
+                f"{resource_group_name}/providers/Microsoft.Compute/virtualMachines"
+                f"/{node.name}"
+            ),
+        }
+
+        protected_settings = {
+            "storageAccountName": storage_account_name,
+            "storageAccountKey": account_credential.get("account_key"),
+        }
+
+        extension_result = extension.create_or_update(
+            name="AzurePerformanceDiagnosticsLinux",
+            publisher="Microsoft.Azure.Performance.Diagnostics",
+            type_="AzurePerformanceDiagnosticsLinux",
+            type_handler_version="1.0",
+            auto_upgrade_minor_version=True,
+            settings=settings,
+            protected_settings=protected_settings,
+        )
+
+        assert_that(extension_result["provisioning_state"]).described_as(
+            "Expected the extension to succeed"
+        ).is_equal_to("Succeeded")
+
+        # Verify report was created and uploaded to the storage account
+        self._container_client = get_or_create_storage_container(
+            credential=platform.credential,
+            subscription_id=platform.subscription_id,
+            cloud=platform.cloud,
+            account_name=storage_account_name,
+            container_name="azdiagextnresults",
+            resource_group_name=resource_group_name,
+        )
+
+        blob_iter = self._container_client.list_blobs(
+            name_starts_with="PerformanceDiagnostics"
+        )
+
+        report_exists = False
+        for _ in blob_iter:
+            report_exists = True
+
+        assert_that(report_exists).is_true()
+
+        # Delete VM Extension
+        extension.delete("AzurePerformanceDiagnosticsLinux")
+
+        assert_that(
+            extension.check_exist("AzurePerformanceDiagnosticsLinux")
+        ).is_false()
+
+    def _is_supported_linux_distro(self, node: Node) -> bool:
+        supported_major_versions = {
+            "redhat": [7, 8],
+            "red hat": [7, 8],
+            "centos": [6, 7],
+            "oracle": [6, 7],
+            "debian": [8, 9, 10, 11],
+            "ubuntu": [14, 16, 18, 20],
+            "suse": [12, 15],
+            "sles": [12, 15],
+            "almalinux": [8],
+            "mariner": [2],
+        }
+
+        version_list = supported_major_versions.get(
+            (node.os.information.vendor).lower()
+        )
+
+        if (
+            version_list is not None
+            and node.os.information.version.major in version_list
+            and not self._is_unsupported_version(node)
+        ):
+            return True
+        else:
+            return False
+
+    def _is_unsupported_version(self, node: Node) -> bool:
+        """
+        These are specific Linux distro versions that are
+        not supported by Azure Performance Diagnostics,
+        even though the major version is generally supported
+        """
+        unsupported_versions = {
```

Comment:
The distro type can be compared as `isinstance(node.os, RedHat)`. It can address derived OS as well.